src/components/image-upload/image-upload.tsx

```diff
@@ -0,0 +1,314 @@
+          ...prevState,
+          image: false
+        }
+      });
+      const fileReader: FileReader = new FileReader();
+      fileReader.onload = () => {
+        if (fileReader.result) {
+          setPickedImage(fileReader.result as string);
+        }
+      };
+      fileReader.readAsDataURL(file);
+    }
+  }
+  const handlePickClick = () => {
+    if (imageInput.current !== null) {
+      imageInput.current.click();
+    }
+  }
+
+  const handleClose = () => {
+    setIsSubmitting(true);
+  };
+
+  const resetForm = () => {
+    if (formRef.current !== null) {
+      formRef.current.reset();
+    }
+    setPickedImage(undefined);
+    setAuthorCheckbox(true);
+    setFormData(initialFormState);
+    setFormError(INITIAL_ERROR_STATE);
+  };
+
+  const handleAuthorCheckboxClick = (value: boolean): void => {
+    setAuthorCheckbox(value);
+    if (user !== undefined) {
+      setFormData({ ...formData, author: user?.fullName });
+      setFormError((prevState) => {
+        const newState = {
+          ...prevState,
+          author: false
+        };
+        return newState;
+      });
+    }
+
+  }
+
+  const onFileUploadProgressChanged = (percentage: number) => {
+    setPercentage(percentage);
+  }
+
+  useEffect(() => setError(""), [logDialogIsOpen]);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      setError("");
+      const createImage = async (): Promise<void> => {
+        try {
+          const fd = new FormData();
+          fd.append('author', formData.author);
+          fd.append('title', formData.title);
+          fd.append('image', formData.image as Blob);
+          fd.append('entityId', formData.entityId);
+          fd.append('entityType', formData.entityType);
+          await createImageAction(token ?? '', fd, onFileUploadProgressChanged);
+          setLogDialogIsOpen(false);
+          resetForm();
+
+          router.refresh();
+        } catch (error) {
+          setError("Prišlo je do napake pri shranjevanju fotografije.");
+          resetForm();
+        }
+        setIsSubmitting(false);
+      }
+
+      const file: File | null = formData.image;
+      const title: string = formData.title;
+      const formAuthor: string = formData.author;
+
+      if (file === null || title.length == 0 || (formAuthor !== null && formAuthor.length == 0)) {
+        setFormError((prevState) => {
+          const newState = {
+            ...prevState,
+            image: file === null,
+            title: title.length == 0,
+            author: formAuthor !== null && formAuthor.length == 0
+          };
+          return newState;
+        });
+        setIsSubmitting(false);
+      } else {
+        if (formRef.current !== null) {
+          formRef.current.reset();
+        }
+        if (imageInput.current) {
+          imageInput.current.value = "";
+        }
+
+        createImage();
+
+
+      }
+    }
+
+  }, [isSubmitting])
+
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
```

Comment:
This does not happen because there is no button in the form. 
You could add a hidden button to trigger this.

Reply:
formRef is not accessible from Dialog callback, therefore I am using isSubmitting variable. I will remove this function as it is not used.